src/neo/SmartContract/ApplicationEngine.Storage.cs

```diff
@@ -204,7 +208,8 @@ protected internal void Put(StorageContext context, byte[] key, byte[] value)
         /// <param name="key">The key of the entry.</param>
         protected internal void Delete(StorageContext context, byte[] key)
         {
-            if (context.IsReadOnly) throw new ArgumentException(null, nameof(context));
+            if (context.IsReadOnly)
```

Comment:
I don't think ANY exception in methods that modify contract storage should be catchable